Prose/ProseLib/Sources/ProseUI/ReactionPicker.swift

```diff
@@ -0,0 +1,245 @@
+    content
+      .compositingGroup()
+      .background {
+        if self.isSelected {
+          RoundedRectangle(cornerRadius: 8)
+            .strokeBorder(.selection, lineWidth: 2)
+            // Make sure the border looks good on material backgrounds
+            // (otherwise not enough contrast)
+            .blendMode(.plusDarker)
+        }
+      }
+      .accessibilityAddTraits(self.isSelected ? .isSelected : [])
+  }
+}
+
+public extension View {
+  func reactionPicker<Action>(
+    store: Store<ReactionPickerState?, Action>,
+    action: @escaping (ReactionPickerAction) -> Action,
+    dismiss: Action
+  ) -> some View {
+    WithViewStore(store) { viewStore in
+      self.popover(unwrapping: viewStore.binding(send: dismiss)) { $state in
+        ReactionPicker(store: store.scope(
+          state: { _ in $state.wrappedValue },
+          action: action
+        ))
+      }
+    }
+  }
+}
+
+struct ReactionPicker_Previews: PreviewProvider {
+  private struct Preview1: View {
+    struct State: Equatable {
+      var reactionPicker: ReactionPickerState?
+      var text: String = ""
+    }
+
+    enum Action: Equatable {
+      case reactionPicker(ReactionPickerAction)
+    }
+
+    static let reducer = Reducer<State, Action, Void>.combine([
+      reactionPickerTogglingReducer.optional().pullback(
+        state: \.reactionPicker,
+        action: CasePath(Action.reactionPicker),
+        environment: { $0 }
+      ),
+      Reducer { state, action, _ in
+        switch action {
+        case let .reactionPicker(.select(reaction)):
+          state.text = reaction.rawValue
+          return .none
+        case .reactionPicker(.deselect):
+          state.text = ""
+          return .none
+        }
+      },
+    ])
+
+    let store: Store<State, Action>
+
+    init(initialState: ReactionPickerState) {
+      self.store = Store(
+        initialState: State(reactionPicker: initialState),
+        reducer: Self.reducer,
+        environment: ()
+      )
+    }
+
+    var body: some View {
+      VStack(alignment: .leading) {
+        WithViewStore(self.store.scope(state: \.text)) { viewStore in
+          Text("Reaction: '\(viewStore.state)'")
+            .padding([.horizontal, .top])
+        }
+        IfLetStore(
+          self.store.scope(state: \.reactionPicker, action: Action.reactionPicker),
+          then: ReactionPicker.init(store:)
+        )
+      }
+      .fixedSize()
+      .previewDisplayName("As a view")
+    }
+  }
+
+  private struct Preview2: View {
+    struct State: Equatable {
+      var reactionPicker: ReactionPickerState?
+      var text: String = ""
+      @BindableState var isShowingPopover: Bool = false
+    }
+
+    enum Action: Equatable, BindableAction {
+      case reactionPicker(ReactionPickerAction)
+      case binding(BindingAction<State>)
+    }
+
+    static let reducer = Reducer<State, Action, Void>.combine([
+      //      reactionPickerTogglingReducer.optional().pullback(
```

Comment:
This can probably be removed?

Reply:
Yes, I just used it from time to time to check that it works in a popover, but didn't wan to add a completely new preview. I could do it, reusing most of the code, but I found this easier